app/controllers/repos_controller.rb

```diff
@@ -138,4 +160,56 @@ def find_benchmark_type_by_category(category)
   def fetch_categories
     @repo.benchmark_types.pluck(:category)
   end
+
+  # Generate the JSON representation of `charts`
+  def generate_json(charts, versions)
+    charts.map do |chart|
```

Comment:
will each entry have 2 entries? If so, we can do this:
` charts.map do |result_data, result_type|`